zlib-rs/src/deflate/algorithm/fast.rs

```diff
@@ -9,10 +9,6 @@ use crate::{
 };
 
 pub fn deflate_fast(stream: &mut DeflateStream, flush: DeflateFlush) -> BlockState {
-    let mut bflush; /* set if current block must be flushed */
-    let mut dist;
-    let mut match_len = 0;
```

Comment:
Historically these variables were pre-declared as an optimization I think. But I do buy that it's not needed any more and in fact having the scope be more restricted might help in re-using registers/stack slots.